test/upgrade/create-in-latest_version-csv-with-header.td

```diff
@@ -0,0 +1,24 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ file-append path=upgrade-csv-with-headers.csv
+id,value
+1,person
+
+> CREATE MATERIALIZED SOURCE csv_upgrade
+  FROM FILE '${testdrive.temp-dir}/upgrade-csv-with-headers.csv'
+  FORMAT CSV WITH HEADER
+
+> CREATE MATERIALIZED SOURCE csv_upgrade_explicit
+  FROM FILE '${testdrive.temp-dir}/upgrade-csv-with-headers.csv'
+  FORMAT CSV WITH HEADER (id, value)
+
+> CREATE MATERIALIZED SOURCE csv_upgrade_no_header
+  FROM FILE '${testdrive.temp-dir}/upgrade-csv-with-headers.csv'
+  FORMAT CSV WITH 2 COLUMNS
```

Comment:
Maybe I don't understand this new upgrade framework, but I don't understand what this test is testing. Doesn't it need to create these sources in v0.8.1, or some other old version that doesn't contain the new code?

Reply:
oh, yeah that makes sense, I was guarding against future breakage not ensuring that I didn't break something from that past.